client/core/types/block.go

```diff
@@ -352,6 +353,11 @@ func (b *Block) AsFragments(size int) (*BlockFragments, error) {
 	return NewDataSetFromData(rawBlock, size), nil
 }
 
+// IsEqual returns true if blocks have same hashes and block numbers
```

Comment:
I'm really surprised ethereum didn't have this. I can only imagine how many comparisons exist in code base

Prob the hash comparison was enough

Reply:
I was surprised too :)

To be sure for 100% lets check hash and block number.